pkg/providers/openstack/stage.go

```diff
@@ -1,10 +1,201 @@
 package openstack
 
 import (
+	"fmt"
+	"github.com/Sirupsen/logrus"
 	"github.com/emc-advanced-dev/pkg/errors"
+	unikos "github.com/emc-advanced-dev/unik/pkg/os"
 	"github.com/emc-advanced-dev/unik/pkg/types"
+	"github.com/rackspace/gophercloud"
+	"github.com/rackspace/gophercloud/openstack/compute/v2/flavors"
+	"github.com/rackspace/gophercloud/openstack/imageservice/v2/images"
+	"github.com/rackspace/gophercloud/pagination"
+	"math"
+	"os"
+	"time"
 )
 
 func (p *OpenstackProvider) Stage(params types.StageImageParams) (_ *types.Image, err error) {
-	return nil, errors.New("not yet supportded for openstack", nil)
+	imageList, err := p.ListImages()
+	if err != nil {
+		return nil, errors.New("failed to retrieve image list", err)
+	}
+
+	// Handle image name collision.
+	for _, image := range imageList {
+		if image.Name == params.Name {
+			if !params.Force {
+				return nil, errors.New(fmt.Sprintf("an image already exists with name '%s', try again with --force", params.Name), nil)
+			} else {
+				logrus.WithField("image", image).Warnf("force: deleting previous image with name '%s'", params.Name)
+				err = p.DeleteImage(image.Id, true)
+				if err != nil {
+					return nil, errors.New("failed to remove existing image", err)
+				}
+			}
+		}
+	}
+
+	clientGlance, err := p.newClientGlance()
+	if err != nil {
+		return nil, err
+	}
+	clientNova, err := p.newClientNova()
+	if err != nil {
+		return nil, err
+	}
+
+	logrus.WithFields(logrus.Fields{
+		"params": params,
+	}).Info("creating boot image from raw image")
+
+	rawImageFile, err := os.Stat(params.RawImage.LocalImagePath)
+	if err != nil {
+		return nil, errors.New("statting raw image file", err)
+	}
+
+	// TODO: Obtain image LOGICAL size, not actual (e.g. 10GB for OSv, not 8MB)
+	imageSizeB := rawImageFile.Size()
```

Comment:
@ilackarms When we run image on OpenStack, we need to pick *flavor* that is big enough (talking HDD-wise right now) for it to run. 

In the case of OSv QEMU image, the physical size is usually small, around 100MB (8MB for OSv core + 80MB for Java + 10kB for application code). By physical size I mean the size that you get when you `ls` or  `stat` the file.

BUT when you compose OSv image using `capstan` tool (in [this line](https://github.com/emc-advanced-dev/unik/blob/master/containers/compilers/osv/java-compiler/osv_boot_creator.go#L124)), you specify a logical size for it. Logical size is the size that OSv sees - so OSv will expand up to this size. The default logical size that capstan uses (if no `--size` argument is passed, as in your case) is 10 GB.

Ofcorse, the flavor must be big enough for the **logical** size, i.e. 10GB in this case, to fit in.


Q: is this size stored somewhere ATM? I couldn't find it so I've added [MinInstanceDiskMB](https://github.com/emc-advanced-dev/unik/pull/78/commits/3f645c9ade360d8bdc7cb6167c89eb99c3688ecb#diff-34dfcb9f9c18e85cfeb1525b94b82572R135), but non of the compilers set it yet (will come with the next PR). This line here should then just read the `MinInstanceDiskMB` field instead of using `stat` to get physical image size.